lambdas/build-client-oauth-response/src/main/java/uk/gov/di/ipv/core/buildclientoauthresponse/BuildClientOauthResponseHandler.java

```diff
@@ -124,6 +124,9 @@ public Map<String, Object> handleRequest(JourneyRequest input, Context context)
                         HttpStatus.SC_BAD_REQUEST, ErrorResponse.MISSING_SESSION_ID);
             }
 
+            ipvSessionItem.setFeatureSet(featureSet);
+            sessionService.updateIpvSession(ipvSessionItem);
```

Comment:
So the idea of this change is that we set the featureset on the session as we're building the OAuth response, then it'll be available to the two other endpoints that Orchestration will ultimately call (issue access token and build user identity), because they're called by orchestration (or its stub), rather than core-front?

Given that this introduces a slight inconsistency between stages in the journey, I'm thinking we should either:
- Make it work more uniformly - i.e. we update the featureset on the session _every_ time we kick off a lambda that is passed a feature set, and use that as the source of truth for the config service.
- Add a comment somewhere (perhaps on `IpvSessionItem`?) that says we _only_ set this at the very end of the journey - i.e. it's not suitable for generally working out what featureset is active.

I don't hugely mind which :)

Reply:
The problem with the first option is that we have to retrieve the session in a lambda before we can make use of a feature set. If we carry on doing what we do we can set it right at the start of the lambda, before we do anything to get the session. That might be useful if we have features that require doing stuff before we get the session.

I'll stick a comment in.